frontend/src/components/Chat/ChatComponent.tsx

```diff
@@ -0,0 +1,52 @@
+import {useEffect, useState} from "react";
+import UserDropdown from "../UsersList/UsersDropDown.tsx";
+import User from "../../models/user.ts";
+import Chat from "./Chat.tsx";
+import {useUser} from "../../context/userContext.tsx";
+import {UsersService} from "../../services/usersService.ts";
+
+interface IMessage {
+    senderId: string;
+    message: string;
+}
+const ChatComponent = () => {
+    const [selectedUser, setSelectedUser] = useState<User | null>(null);
+    const [messages, setMessages] = useState<IMessage[]>([]);
+    const [users, setUsers] = useState<User[]>([]);
+    const { user, setUser } = useUser();
+    const senderId =JSON.parse(sessionStorage.user)._id;
+
+    useEffect(() => {
+            const usersService = new UsersService(user!, setUser);
+            const { request } = usersService.getAllUsers();
+            request
+                .then((response) => {
+                    setUsers(response.data);
+                })
+                .catch((err) => {
+                    console.error(err);
+                });
+    });
```

Comment:
```suggestion
    }, [user]);
```
user is a dependency for this effect, moreover without any dependency (even an empty array) the effect runs endlessly